docs/src/main/asciidoc/panache-rest.adoc

```diff
@@ -0,0 +1,148 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+
+= Quarkus - Generating CRUD resources with Panache REST
+
+include::./attributes.adoc[]
+:extension-status: experimental
+
+A lot of web applications are monotonous CRUD applications with boring to write REST APIs.
+To ease this task Panache REST can generate the basic CRUD endpoints for your entities and repositories.
+
+While this extension is still experimental and provides a limited feature set, we hope to get an early feedback for it.
+Currently this extension supports Hibernate ORM with Panache and can generate CRUD resources that work with `application/json` and `application/hal+json` content.
+
+include::./status-include.adoc[]
+
+== Setting up Panache REST
+
+To begin with:
+
+* Add the required dependencies to your `pom.xml`
+** Panache REST with Hiberante ORM extension (`quarkus-panache-rest-hibernate-orm`)
+** A JDBC driver extension (`quarkus-jdbc-postgresql`, `quarkus-jdbc-h2`, `quarkus-jdbc-mariadb`, ...)
+** One of the RestEasy JSON serialization extensions (`quarkus-resteasy-jackson` or `quarkus-resteasy-jsonb`)
+
+[source,xml]
+----
+<dependencies>
+    <!-- Panache REST with Hibernate ORM -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-panache-rest-hibernate-orm</artifactId>
+    </dependency>
+
+    <!-- JDBC driver dependencies -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-jdbc-postgresql</artifactId>
+    </dependency>
+
+    <!-- JSON serializer dependency -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-resteasy-jackson</artifactId>
+    </dependency>
+</dependencies>
+----
+
+* Implement the Panache entities and/or repositories as explained in the link:hibernate-orm-panache[Panache with Hibernate ORM guide].
+* Define the interfaces for generation as explained in the sections below.
+
+== Generating CRUD resources
+
+Panache REST generates CRUD resources based on the interfaces available in your application.
+For each entity and repository that you want to generate, provide an interface that extends either `PanacheEntityCrudResource` or `PanacheRepositoryCrudResource` interface.
+_Do not implement these interfaces and don't provide custom methods because they won't be implemented._
+You can, however, override the methods from the extended interface in order to customize them (see the section at the end).
+
+The resources that will be generated will be functionally equivalent for both entities and repositories.
+Only the data will accessed using either one or the other.
+The skeleton of the generated JAX-RS resource will look like this:
+
+[source,java]
```

Comment:
I think we need a little more emphasis that on the fact this is generated - not something the user would write.
Mentioning it because looking at the code initially through me off...

Reply:
It does say "generated JAX-RS resource". Any suggestion?

Reply:
Yeah I know, it's just that it's too easy to think you need to write this. 
Perhaps change the name of the class and add a comment on it?

Reply:
I've changed the sentence above